Dockerfile.multi-stage

```diff
@@ -1,10 +1,11 @@
 FROM golang:1.13.6-alpine3.11 as builder
 
-ADD . $GOPATH/src/github.com/thanos-io/thanos
 WORKDIR $GOPATH/src/github.com/thanos-io/thanos
 
 RUN apk update && apk upgrade && apk add --no-cache alpine-sdk
 
+ADD . $GOPATH/src/github.com/thanos-io/thanos
```

Comment:
Consider adding a comment why would this improve docker layer caching, so that in future we know to leave it here.

Reply:
Thanks for reviewing, I have added the comments.